testsuite/manualmode/src/test/java/org/jboss/as/test/manualmode/auditlog/AbstractLogFieldsOfLogTestCase.java

```diff
@@ -0,0 +1,154 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jboss.as.test.manualmode.auditlog;
+
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.AUTHENTICATION;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.CORE_SERVICE;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.LOCAL;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.MANAGEMENT;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.SECURITY_REALM;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Pattern;
+
+import org.jboss.as.controller.client.ModelControllerClient;
+import org.jboss.as.controller.client.Operation;
+import org.jboss.as.controller.client.OperationBuilder;
+import org.jboss.as.controller.client.helpers.Operations;
+import org.jboss.as.controller.client.helpers.Operations.CompositeOperationBuilder;
+import org.jboss.as.test.shared.TestSuiteEnvironment;
+import org.jboss.dmr.ModelNode;
+import org.junit.Assert;
+import org.xnio.IoUtils;
+
+/**
+ * @author <a href="mailto:jperkins@redhat.com">James R. Perkins</a>
+ */
+abstract class AbstractLogFieldsOfLogTestCase {
+
+    private final String defaultUserKey = "wildfly.sasl.local-user.default-user";
+    private ModelNode userAuthAddress;
+    private ModelNode userIdentRealmAddress;
+    private boolean elytronEnabled = false;
+
+    final Pattern dateStampPattern = Pattern.compile("\\d\\d\\d\\d-\\d\\d-\\d\\d \\d\\d:\\d\\d:\\d\\d - \\{");
+    final Path file = Paths.get(TestSuiteEnvironment.getJBossHome(), "standalone", "data", "audit-log.log");
+
+    void configureUser(final ModelControllerClient client, final CompositeOperationBuilder compositeOp) throws IOException {
+
+        // Determine the we should be using a security-realm or SASL (we assume SASL means Elytron is enabled)
+        String securityRealm = "ManagementRealm";
+        ModelNode op = Operations.createReadResourceOperation(Operations.createAddress("core-service", "management", "management-interface", "http-interface"));
+        ModelNode result = executeForSuccess(client, OperationBuilder.create(op).build());
+        if (result.hasDefined("security-realm")) {
```

Comment:
I'm still a tiny bit concerned about using this as the mechanism to check if elytron is enabled or not, or if both legacy and elyton are mixed or something (Don't know if this is even realistic.) I think this is probably OK here though, since we know whats in the configs.